src/main/java/com/iotiq/application/domain/authority/OrderedProductManagementAuthority.java

```diff
@@ -0,0 +1,16 @@
+package com.iotiq.application.domain.authority;
+
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.stereotype.Component;
+
+@Component("OrderedProductManagementAuth")
+public class OrderedProductManagementAuthority {
```

Comment:
Why is this class necessary if the granted authority values are the same as the user management values?

Reply:
We add it to the GrantedAuthority List in the Role class

COMPANY(
            new SimpleGrantedAuthority("ROLE_COMPANY"),
            ProductDemandManagementAuthority.VIEW,

Reply:
The values in this class produce the same SimpleGrantedAuthority as all the other granted authority classes:

```
    public static final GrantedAuthority VIEW = new SimpleGrantedAuthority("USER_MANAGEMENT_VIEW");
    public static final GrantedAuthority UPDATE = new SimpleGrantedAuthority("USER_MANAGEMENT_UPDATE");
```

If you also have this ☝️ in a class called UserManagementAuthority and use them to give different roles different authorities that wont work like you expect them to work because the values of the granted authorities is the same, no?

What I mean is these two are the same:

```
@PreAuthorise(hasAuthority(OrderedProductManagementAuthority.VIEW))
@PreAuthorise(hasAuthority(UserManagementAuthority.VIEW))
```

Because both have the value `USER_MANAGEMENT_VIEW`

Reply:
I opened an issue for this, i will do it there
https://github.com/Iotiq-Erste/kimono-application/issues/47